### ISSUE TITLE
sqlalchemy error 'server closed the connection unexpectedly' while running network creation code

### ISSUE THREAD

**Issue by carlhiggs** [healthysustainablecities/global-indicators#212]
While processing a new test example for Melbourne, I experienced this apparent server timeout related error while running the _03_create_network_resources.py subprocess:

```
Traceback (most recent call last):
  File "/env/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1900, in _execute_context
    self.dialect.do_execute(
  File "/env/lib/python3.10/site-packages/sqlalchemy/engine/default.py", line 736, in do_execute
    cursor.execute(statement, parameters)
psycopg2.OperationalError: server closed the connection unexpectedly
	This probably means the server terminated abnormally
	before or while processing the request.


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/ghsci/work/process/subprocesses/_03_create_network_resources.py", line 256, in <module>
    main()
  File "/home/ghsci/work/process/subprocesses/_03_create_network_resources.py", line 145, in main
    db_contents.has_table('nodes') and db_contents.has_table('edges')
  File "/env/lib/python3.10/site-packages/sqlalchemy/engine/reflection.py", line 283, in has_table
    return self.dialect.has_table(conn, table_name, schema)
  File "/env/lib/python3.10/site-packages/sqlalchemy/dialects/postgresql/base.py", line 3596, in has_table
    cursor = connection.execute(
  File "/env/lib/python3.10/site-packages/sqlalchemy/future/engine.py", line 280, in execute
    return self._execute_20(
  File "/env/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1705, in _execute_20
    return meth(self, args_10style, kwargs_10style, execution_options)
  File "/env/lib/python3.10/site-packages/sqlalchemy/sql/elements.py", line 334, in _execute_on_connection
    return connection._execute_clauseelement(
  File "/env/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1572, in _execute_clauseelement
    ret = self._execute_context(
  File "/env/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1943, in _execute_context
    self._handle_dbapi_exception(
  File "/env/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 2124, in _handle_dbapi_exception
    util.raise_(
  File "/env/lib/python3.10/site-packages/sqlalchemy/util/compat.py", line 211, in raise_
    raise exception
  File "/env/lib/python3.10/site-packages/sqlalchemy/engine/base.py", line 1900, in _execute_context
    self.dialect.do_execute(
  File "/env/lib/python3.10/site-packages/sqlalchemy/engine/default.py", line 736, in do_execute
    cursor.execute(statement, parameters)
sqlalchemy.exc.OperationalError: (psycopg2.OperationalError) server closed the connection unexpectedly
	This probably means the server terminated abnormally
	before or while processing the request.

[SQL: select relname from pg_class c join pg_namespace n on n.oid=c.relnamespace where pg_catalog.pg_table_is_visible(c.oid) and relname=%(name)s]
[parameters: {'name': 'nodes'}]
(Background on this error at: https://sqlalche.me/e/14/e3q8)
```

The last message before the error was the following:
```
2023-03-13 17:34:29 Saved graph as GraphML file at "/home/ghsci/work/process/data/_study_region_outputs/melbourne_au_2023_au_2020/melbourne_au_2023_au_2020_1600m_pedestrian_osm_20230307.graphml"
Done.
```

This indicates that the error was occurring around here:

https://github.com/global-healthy-liveable-cities/global-indicators/blob/7c7b974b2c0b47dba20646c3d9c0e77a5b0b0b93/process/subprocesses/_03_create_network_resources.py#L144-L146

The `db_contents` variable is defined much earlier in the code and contains the results of running the [inspect() function](https://docs.sqlalchemy.org/en/20/core/inspection.html).  Perhaps this timesout if not accessed for some time? (e.g. Melbourne network is larger than some cities, so may take longer).   I didn't think that was the case however (I thought it was an in-memory object, not accessing the sql database itself).

It seems when the code is re-run following this it proceeds successfully.   

Perhaps inspect should be used directly, rather than attempt to save it in a variable

Anyway - not exactly sure what is causing this issue that doesn't seem to appear for smaller cities, but has appeared for me in a couple of Melbourne tests now (so far dealt with by re-running, which uses the saved results and presumably avoids the timeout; but we should sort out underlying issue)

**Comment by carlhiggs** [comment]
Using the additional connection parameters described in [this StackOverflow post](https://stackoverflow.com/a/67421581/4636357) appears to resolve this issue, avoiding the timeout:
```{python}
create_engine(
        connection_string,
        pool_pre_ping=True,
        connect_args={
            "keepalives": 1,
            "keepalives_idle": 30,
            "keepalives_interval": 10,
            "keepalives_count": 5,
        }
    )
```

I've re-run this with Melbourne where this timeout was observed, and the place where a timeout reliably seemed to occur for that city no longer times out

**Comment by carlhiggs** [comment]
This was addressed in #218
